src/bin/tui/table.rs

```diff
@@ -616,7 +616,7 @@ impl<T: TableViewItem<H>, H: Eq + Hash + Copy + Clone + 'static> TableView<T, H>
 			});
 			self.rows_to_items = rows_to_items;
 
-			self.set_selected_item(old_item);
+			old_item.map(|o| self.set_selected_item(o));
```

Comment:
Oh that's really nice - just using `map()` rather than an `if let Some(x) = old_item { ... }`.